src/cpp/rtps/builtin/discovery/endpoint/EDPServer.hpp

```diff
@@ -104,6 +104,30 @@ class EDPServer : public fastrtps::rtps::EDPSimple
     bool removeLocalWriter(
             fastrtps::rtps::RTPSWriter* W) override;
 
+    /**
+     * This method removes all changes from the correct data writer history with the same identity as the one in the disposal_change
+     * and adds it to the corresponding builtin EDP endpoint for forwarding it to interested remotes.
+     * @param change Pointer to the CacheChange_t.
+     * @param discovery_db Reference to the discovery DB.
+     * @param change_guid_prefix The identity of the participant from which the change came.
+     * @return True if successful.
+     */
+    bool process_disposal(
+            fastrtps::rtps::CacheChange_t* disposal_change,
+            fastdds::rtps::ddb::DiscoveryDataBase& discovery_db,
+            fastrtps::rtps::GuidPrefix_t& change_guid_prefix);
+
+    /**
+     * This method removes all changes from the correct data writer history (change->witerGUID.enitity)
+     * and releases the change from the correct history dependeing whether the change is from
+     * this participant or remote
+     * @param change Pointer to the CacheChange_t.
+     * @return True if successful.
+     */
+    bool process_and_release_change(
+            fastrtps::rtps::CacheChange_t* change,
+            const bool& is_remote_participant = false);
```

Comment:
No need for a const reference here:

```suggestion
            bool is_remote_participant = false);
```